grpc_test.go

```diff
@@ -0,0 +1,91 @@
+package openmock
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"google.golang.org/grpc"
+
+	"github.com/checkr/openmock/demo_protobuf"
+)
+
+// arbitraryPort returns a non-used port.
+func arbitraryPort() int {
+	l, err := net.Listen("tcp", "127.0.0.1:0")
+	if err != nil {
+		panic(err)
+	}
+
+	defer l.Close()
+	addr := l.Addr().String()
+	_, port, err := net.SplitHostPort(addr)
+	if err != nil {
+		panic(err)
+	}
+	p, _ := strconv.Atoi(port)
+	return p
+}
+
+var (
+	testGRPCHost       = "127.0.0.1"
+	testGRPCPort       = arbitraryPort()
+	demoGRPCServiceMap = map[string]GRPCService{
```

Comment:
moved this demo as an example in the `_test.go` file so that it won't be compiled to normal code.